src/main/python/pybuilder/plugins/python/distutils_plugin.py

```diff
@@ -312,32 +310,144 @@ def build_scripts_string(project):
+        returnString += (" " * (indent+4)) + "('%s', ['" % dataType
+        returnString += "', '".join(dataFiles)
+        returnString += "']),\n"
 
-    return "data_files = %s," % str(data_files)
+    returnString = returnString[:-2] + "\n"
+    returnString += " " * indent + "]"
+    return returnString
+    return build_string_from_array(data_files)
 
 
 def build_package_data_string(project):
+    indent = 8
+
+    sorted_keys = sorted(project.package_data.keys())
+
     package_data = project.package_data
     if package_data == {}:
-        return ""
-    package_data_string = "package_data = {"
+        return "{}"
+
+    returnString = "{\n"
+
+    for pkgType in sorted_keys:
+        returnString += " " * (indent+4)
+        returnString += "'%s': " % pkgType
+        returnString += "['"
+        returnString += "', '".join(package_data[pkgType])
+        returnString += "'],\n"
+
+    returnString = returnString[:-2] + "\n"
+    returnString += " " * indent + "}"
+
+    return returnString
+
+    return build_string_from_dict(package_data)
+
+
+def build_packages_string(project):
+    pkgs = [pkg for pkg in project.list_packages()]
+    if len(pkgs) > 0:
+        return build_string_from_array(pkgs)
+    else:
+        return '[]'
+
+
+def build_modules_string(project):
+    mods = [mod for mod in project.list_modules()]
+    if len(mods) > 0:
+        return build_string_from_array(mods)
+    else:
+        return '[]'
+
+
+def build_console_scripts_string(project):
+    console_scripts = project.get_property('distutils_console_scripts', [])
+
+    if len(console_scripts) == 0:
+        return "{}"
+
+    indent = 12
+    string = "{'console_scripts': "
+    string += build_string_from_array(console_scripts, indent)
+    string += "}"
+
+    return string
+
+
+def build_classifiers_string(project):
+    classifiers = project.get_property('distutils_classifiers', [])
+    return build_string_from_array(classifiers, indent=12)
+
+
+def build_string_from_array(arr, indent=12):
+    returnString = ""
+
+    if len(arr) == 1:
+        if not isinstance(arr[0], (frozenset, list, set, tuple)):
+            if len(arr[0]) > 0:
+                returnString += "['%s']" % arr[0]
+        else:
+            if len(arr[0]) > 0:
+                returnString += "[" + build_string_from_array(arr[0], indent+4) + "]"
+    elif len(arr) > 1:
+        returnString = "[\n"
+
+        for item in arr:
+            if len(item) > 0:
+                if not isinstance(item, (frozenset, list, set, tuple)):
+                    returnString += (" " * indent) + "'" + item + "',\n"
+            else:
+                returnString += (" " * indent) + build_string_from_array(item, indent+4) + ",\n"
+
+        returnString = returnString[:-2] + "\n"
+        returnString += " " * (indent - 4)
+        returnString += "]"
+
+    return returnString
+
+
+def build_string_from_dict(d, indent=12):
+    mapStrings = []
+
+    print d
```

Comment:
yeah, I've noticed. I'll have to do some extra testing with python3. But otherwise, worth the effort?